### PR TITLE
upodate Metabase CVE-2021-41277 detector

### PR DESCRIPTION
**Pull request by C4o**
Hi @magl0 , cause #197 PR is nearly done and this vulnerability is published in 2 weeks and it's score is critical, so I complete the detector of CVE-2021-41277, please review it. Thanks!

isseue: #208 

vulnerable docker
```
docker run -d -p 80:3000 metabase/metabase:v0.40.4
```

fixed docker
```
docker run -d -p 80:3000 metabase/metabase:v0.40.5
```

### CODE REVIEW

File: community/README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ This directory contains plugins contributed by community members.
 #### Path Traversal
 *   [Apache HTTP Server Path traversal and disclosure (CVE-2021-41773) Detector](https://github.com/google/tsunami-security-scanner-plugins/tree/master/community/detectors/apache_http_server_cve_2021_41773)
 *   [Apache Solr RemoteStreaming Arbitrary File Reading Detector](https://github.com/google/tsunami-security-scanner-plugins/tree/master/community/detectors/apache_solr_arbitrary_file_reading)
+*   [MetaBase Local_File_Inclusion Detector](https://github.com/google/tsunami-security-scanner-plugins/tree/master/community/detectors/metabase_cve_2021_41277)
```
Comment (magl0) [inline community/README.md:14]: Propose to update the link to `Metabase CVE-2021-41277 LFI (Local File Inclusion) Detector`.

File: community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * A {@link VulnDetector} that detects MetaBase Local File Inclusion(CVE-2021-41277)
+ */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "MetaBaseCVE202141277Detector",
+    version = "0.1",
+    description = "This detector checks for MetaBase Local File Inclusion(CVE-2021-41277).",
+    author = "C4o (syttcasd@gmail.com)",
+    bootstrapModule = MetaBaseCVE202141277DetectorBootstrapModule.class)
+public final class MetaBaseCVE202141277Detector implements VulnDetector {
```
Comment (magl0) [inline community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java:57]: [Google Java Code Style: camelCase](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case): please use MetabaseCve202141277Detector` instead.

File: community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                    .setValue("MetaBase_CVE_2021_41277_Local_File_Inclusion"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("CVE-2021-41277")
```
Comment (magl0) [inline community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java:113]: Please use a more descriptive title, e.g. `Metabase CVE-2021-41277 Local File Inclusion Vulnerability`.

File: community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                    .setValue("MetaBase_CVE_2021_41277_Local_File_Inclusion"))
```
Comment (magl0) [inline community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetaBaseCVE202141277Detector.java:111]: Please use `UPPER_SNAKE_CASE` for the ID value.

File: community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetabaseCve202141277Detector.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                    .setValue("Metabase CVE-2021-41277 Local File Inclusion Vulnerability"))
+                .setSeverity(Severity.CRITICAL)
+                .setTitle("CVE_2021_41277")
```
Comment (magl0) [inline community/detectors/metabase_cve_2021_41277/src/main/java/com/google/tsunami/plugins/detectors/metabase/MetabaseCve202141277Detector.java:113]: Sorry for the confusions. I mean `setTitle` to `Metabase CVE-2021-41277 Local File Inclusion Vulnerability` and `MainId.setValue` to `CVE_2021_41277`.

**Review by magl0 — COMMENTED** [review]
This genuinely looks good. I'll start redirecting this to internal code reviews.

**Comment by C4o** [comment]
Hi @magl0 , I've updated it, please review again.

**Comment by magl0** [comment]
There seems to be a merge conflict, can you resolve it?

**Comment by C4o** [comment]
Yes, I've resolved it.

**Comment by magl0** [comment]
Great! I'll redirect this to internal reviews. I'll ping the reviewer for the other PR as well.

**Comment by C4o** [comment]
Thanks! BTW, is there problem with #197 ?

**Comment by magl0** [comment]
As I said, I'll ping the reviewer for #197 :) It will be automatically merged if the reviewer does not have any comments on it. Otherwise I'll mirror them back to the original PR.

**Pull request merged**